### ISSUE TITLE
Support for media.ccc.de 

### ISSUE THREAD

**Issue by TheAssassin** [TeamNewPipe/NewPipeExtractor#105]
I would like to see NewPipe become _the_ Android client for https://media.ccc.de. The C3VOC's media portal supports both static content and live streams.

There's another client for Android, which might be useful for inspiration (maybe the developer would even collaborate in implementing the functionality in NewPipeExtractor): https://github.com/NiciDieNase/chaosflix

**Comment by theScrabi** [comment]
That is a good idea :D help is always welcome.

**Comment by FlorianSteenbuck** [comment]
If we really want to write this compatibility mode this only a database structure which should not implement because we got our own data structures. I have not seen in this project parsing from the media.cc.de website. Maybe it is using the streaming.media.ccc.de api, but that seems to be all to me.

### Problematics of the Implementation

#### Shortly Described
##### Problem
This is not a website that get a database in the back it is website that is orientated on representation of the data. It not carrying much about a norm it is just a viewable chaotic site. We can only parse the document. We got no preloading which is nice, but also no readable data und sometimes they change everything which makes it quit hard to parse the website. The most Implementations create a database from static references and using the database. This is not how a implementation works.

##### Solution
Their are multiple things that we need to do that requires static references, data preparation, unrecoverable detection and for new conferences deep data searching (split SearchEngine out of the StreamingService and add a SearchEngineService) for systems that contain actual real data. The media.ccc.de is only of a few parts from these huge load of bullshit we need to parse.

#### Conference Management
We got at least 4 different systems in multiple version to parse that actually manage these whole conferences with real data management. And the best is we need map a lot. Mapping of User Names and Mapping of Playlist/Series and Mapping for get these Management Systems and Podcast out of the c3voc calendar and this is going to fail because their are multiple management systems out their that are currently simply down or where never setup or out of the relevant range of this calendar even the raw data sources got only things in this range.

#### Video Data
We got Videos like this that simply fail on every data level:
https://media.ccc.de/v/dg-92
And do not expect that any link works fine here and think their got author database, no simply got no author database. And do not expect that the event tags are every time the same. No they use different tag short forms over the years.

#### Need Support
In addition we need (hash-)tag support and directory categories (useful for twitch implementation). We should skip the video cdn and use a filter that for the mirros. Maybe I can post invest more into this plattform and can post again more information on how to implement this plattform. I love to see this plattform in action on NewPipe

**Comment by theScrabi** [comment]
@FlorianSteenbuck maybe you are interested in directly adding support for media.ccc in the extractor.
A documentation about how you can do it can be found here: https://teamnewpipe.github.io/documentation/

**Comment by FlorianSteenbuck** [comment]
@TheAssassin Normally we do not use public apis. According to the latest talk about NewPipe this is because we want to got no contract with the plattform that we integrate. Also for me it is motivation to do it the hard way and that integrate like the user want it. Also their may be a restriction in from the F-Droid that can kick us out of their store. A Solution would be a plugin-able light version of NewPipe, we already use rhino why not using it for plugins.

@theScrabi I doing currently a lot of stuff private and public . If you want to help me with the PeerTube Implementation I got currently a problem with my [Punycode Implementation](https://github.com/FlorianSteenbuck/OtherPunycode). I try to implement this inspired by a idna_decode php script, the [popular javascript implementation](https://github.com/bestiejs/punycode.js/) and the [punycode rfc](https://tools.ietf.org/html/rfc3492). Their are also some featured that might be not needed like the javascript interface for arrays and strings. Punycode is needed for a Safe DNS Lib that can be used for DNS Discovery which is required by STUN and TURN. STUN, TURN, RTP and RTCP are required by WebRTC. WebRTC, Websocket and SDP are required by WebTorrent.

If you are interested in the python change, which would a cool change for a cross-plattform app with pyside (maybe this also can be used for iOS), wait for next commit in [python-generics](https://github.com/FlorianSteenbuck/python-generics), try to understand my code, implement the needed enchantments that are needed for the NewPipeExtractor and rewrite the code in python 2.

**Comment by TheAssassin** [comment]
> Normally we do not use public apis. According to the latest talk about NewPipe this is because we want to got no contract with the plattform that we integrate. Also for me it is motivation to do it the hard way and that integrate like the user want it. Also their may be a restriction in from the F-Droid that can kick us out of their store. A Solution would be a plugin-able light version of NewPipe, we already use rhino why not using it for plugins.

The idea is to protect the users' privacy by not using APIs which require a login. The voctoweb API however is publicly available, and I would trust the CCC it's not monitored. Therefore it's safe to use IMO. @theScrabi what do you think?

**Comment by theScrabi** [comment]
> If you are interested in the python change, which would a cool change for a cross-plattform app with pyside

no python in NewPipe.

@FlorianSteenbuck you may ask the Peertube guys, the leading developer gave us some bounty to make NewPipe compatible with NewPipe. So they might support you.

> Normally we do not use public apis. According to the latest talk about NewPipe this is because we want to got no contract with the plattform that we integrate. 

I didn't say no public apis. Public apis are ok as long as they don't require a contract for getting an access token. I don't think media.ccc is so restrictive, and whenever we have the possibility to use an API we should do it. Because `API > Web`.

**Comment by wb9688** [comment]
Support for media.ccc.de has been implemented
